### PR TITLE
feat: arm64 zk-environment rust Docker images and other

### PR DESCRIPTION
**Pull request by hatemosphere**
# What ❔

- Bump zksolc and zksvyper binaries, build solidity from source (to support arm64 in future)
- Minor CI and Dockerfiles refactoring here and there
- Bump ancient gcloud CLI in zk-env publish CI
- Bump Ubuntu in zk-env to 22.04
- **Build zk-environment (non-CUDA) on arm64**

## Why ❔

To unblock building of the core services on arm64 arch

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: docker/contract-verifier/Dockerfile [hunk 1]
```diff
@@ -63,10 +63,10 @@ RUN mkdir -p /etc/zksolc-bin/v1.3.9 \
     && wget https://github.com/matter-labs/zksolc-bin/raw/main/linux-amd64/zksolc-linux-amd64-musl-v1.3.9 \
     && mv zksolc-linux-amd64-musl-v1.3.9 /etc/zksolc-bin/v1.3.9/zksolc \
     && chmod +x /etc/zksolc-bin/v1.3.9/zksolc
-RUN mkdir -p /etc/zksolc-bin/v1.3.10 \
```
Comment (perekopskiy) [inline docker/contract-verifier/Dockerfile:66]: Please, do not remove old zksolc version from Dockerfile

File: docker/contract-verifier/Dockerfile [hunk 2]
```diff
@@ -75,20 +75,20 @@ RUN mkdir -p /etc/zksolc-bin/v1.3.13 \
     && wget https://github.com/matter-labs/zksolc-bin/raw/main/linux-amd64/zksolc-linux-amd64-musl-v1.3.13 \
     && mv zksolc-linux-amd64-musl-v1.3.13 /etc/zksolc-bin/v1.3.13/zksolc \
     && chmod +x /etc/zksolc-bin/v1.3.13/zksolc
-RUN mkdir -p /etc/zksolc-bin/v1.3.14 \
```
Comment (perekopskiy) [inline docker/contract-verifier/Dockerfile:78]: Same here

File: docker/contract-verifier/Dockerfile [hunk 3]
```diff
@@ -75,20 +75,20 @@ RUN mkdir -p /etc/zksolc-bin/v1.3.13 \
...
+    && wget https://github.com/matter-labs/zksolc-bin/raw/main/linux-amd64/zksolc-linux-amd64-musl-v1.3.15 \
+    && mv zksolc-linux-amd64-musl-v1.3.15 /etc/zksolc-bin/v1.3.15/zksolc \
+    && chmod +x /etc/zksolc-bin/v1.3.15/zksolc
 
 # install zkvyper
 RUN mkdir -p /etc/zkvyper-bin/v1.3.9 \
     && wget https://github.com/matter-labs/zkvyper-bin/raw/main/linux-amd64/zkvyper-linux-amd64-musl-v1.3.9 \
     && mv zkvyper-linux-amd64-musl-v1.3.9 /etc/zkvyper-bin/v1.3.9/zkvyper \
     && chmod +x /etc/zkvyper-bin/v1.3.9/zkvyper
-RUN mkdir -p /etc/zkvyper-bin/v1.3.10 \
```
Comment (perekopskiy) [inline docker/contract-verifier/Dockerfile:88]: Same here

File: docker/contract-verifier/Dockerfile (hunk 1 above)
Comment (hatemosphere) [reply docker/contract-verifier/Dockerfile:66]: ah makes sense, you need all versions to be present, i'll fix this

File: .github/workflows/ci-core-reusable.yml [hunk 4]
```diff
@@ -119,20 +118,20 @@ jobs:
           mv solc-linux-amd64-v0.8.21+commit.d9974bed $(pwd)/etc/solc-bin/0.8.21/solc
           chmod +x $(pwd)/etc/solc-bin/0.8.21/solc
 
-          mkdir -p $(pwd)/etc/zksolc-bin/v1.3.14
```
Comment (alexandrst88) [inline .github/workflows/ci-core-reusable.yml:122]: can we add variable to not change in n places?

**Review by alexandrst88 — CHANGES_REQUESTED** [review]

**Review by otani88 — APPROVED** [review]

File: .github/workflows/ci-core-reusable.yml [hunk 5]
```diff
@@ -217,20 +216,20 @@ jobs:
...
-          wget https://github.com/matter-labs/zksolc-bin/raw/main/linux-amd64/zksolc-linux-amd64-musl-v1.3.14
-          mv zksolc-linux-amd64-musl-v1.3.14 $(pwd)/etc/zksolc-bin/v1.3.14/zksolc
-          chmod +x $(pwd)/etc/zksolc-bin/v1.3.14/zksolc
+          mkdir -p $(pwd)/etc/zksolc-bin/v1.3.15
+          wget https://github.com/matter-labs/zksolc-bin/raw/main/linux-amd64/zksolc-linux-amd64-musl-v1.3.15
+          mv zksolc-linux-amd64-musl-v1.3.15 $(pwd)/etc/zksolc-bin/v1.3.15/zksolc
```
Comment (alexandrst88) [inline .github/workflows/ci-core-reusable.yml:221]: we can just download bin to dir directly without redundant `mv` call

File: .github/workflows/ci-core-reusable.yml (hunk 4 above)
Comment (hatemosphere) [reply .github/workflows/ci-core-reusable.yml:122]: let's takle this in a separate PRs, this one is already too big

**Review by alexandrst88 — CHANGES_REQUESTED** [review]

File: docker/contract-verifier/Dockerfile [hunk 6]
```diff
@@ -22,73 +22,19 @@ FROM debian:bookworm-slim
...
-# install zkvyper
-RUN mkdir -p /etc/zkvyper-bin/v1.3.9 \
-    && wget https://github.com/matter-labs/zkvyper-bin/raw/main/linux-amd64/zkvyper-linux-amd64-musl-v1.3.9 \
-    && mv zkvyper-linux-amd64-musl-v1.3.9 /etc/zkvyper-bin/v1.3.9/zkvyper \
-    && chmod +x /etc/zkvyper-bin/v1.3.9/zkvyper
-RUN mkdir -p /etc/zkvyper-bin/v1.3.10 \
-    && wget https://github.com/matter-labs/zkvyper-bin/raw/main/linux-amd64/zkvyper-linux-amd64-musl-v1.3.10 \
-    && mv zkvyper-linux-amd64-musl-v1.3.10 /etc/zkvyper-bin/v1.3.10/zkvyper \
-    && chmod +x /etc/zkvyper-bin/v1.3.10/zkvyper
+# install zksolc 1.3.0-1.3.15
```
Comment (popzxc) [inline docker/contract-verifier/Dockerfile:25]: Arguably it's not `ci`, but rather `feat`, since we're changing capabilities of an actual application.

**Review by yorik — DISMISSED** [review]

File: sqlx_v0.5.13.patch [hunk 7]
```diff
@@ -0,0 +1,21 @@
+diff --git a/Cargo.toml b/Cargo.toml
```
Comment (popzxc) [inline sqlx_v0.5.13.patch:1]: Should it be there? I guess not.

File: docker/contract-verifier/Dockerfile (hunk 6 above)
Comment (perekopskiy) [inline docker/contract-verifier/Dockerfile:25]: 1.3.12 should be skipped.. this zksolc was buggy and its binaries were removed

Comment (hatemosphere) [reply docker/contract-verifier/Dockerfile:25]: agree

File: .github/workflows/ci-core-reusable.yml [hunk 8]
```diff
@@ -33,7 +33,6 @@ jobs:
           ci_run zk db setup
           ci_run zk compiler all
           ci_run zk contract build
-          ci_run zk run yarn
```
Comment (perekopskiy) [inline .github/workflows/ci-core-reusable.yml:36]: Why is it removed? I'm fine with it If it works, just curious

Comment (hatemosphere) [reply .github/workflows/ci-core-reusable.yml:36]: @perekopskiy https://github.com/matter-labs/zksync-era/pull/296/files#diff-293b35dbbaf364cec87320ca4091f93d974165e35cc640d7f44cfb9f51ebed0dL32 there's a duplicate line above

File: docker/contract-verifier/Dockerfile (hunk 6 above)
Comment (hatemosphere) [reply docker/contract-verifier/Dockerfile:25]: done

File: sqlx_v0.5.13.patch (hunk 7 above)
Comment (hatemosphere) [reply sqlx_v0.5.13.patch]: apparently, dependency is available in crates.io again, removing the hack

**Review by popzxc — APPROVED** [review]

**Review by perekopskiy — DISMISSED** [review]

**Review by perekopskiy — APPROVED** [review]

**Review by yorik — APPROVED** [review]

**Pull request merged**